app/javascript/app/service-provider-form.js

```diff
@@ -54,28 +54,51 @@ function protocolOptionSetup() {
   const samlFields = document.querySelectorAll('.saml-fields');
   const oidcFields = document.querySelectorAll('.oidc-fields');
   const returnToSpUrl = document.getElementById('service_provider_return_to_sp_url');
+  const returnToSpUrlLabel = returnToSpUrl.previousSibling;
+  const abbrElement = document.createElement("ABBR");
+  const requiredText = document.createTextNode("(*required)");
+
+  // Functions //
+
+  // Removes required form validations that were set in simple_form for protocols that don't require the attribute
+  const removeRequiredReturnUrlValidation = () => {
+    returnToSpUrl.removeAttribute('required');
+    returnToSpUrlLabel.classList.remove('usa-input-required', 'required');
+    returnToSpUrlLabel.removeChild(returnToSpUrlLabel.lastElementChild); // this removes the abbr element that contains the "required" text from the field
+  };
+
+  const addRequiredReturnUrlValidation = () => {
+    returnToSpUrl.setAttribute('required', 'required');
+    returnToSpUrlLabel.classList.add('usa-input-required', 'required');
+    abbrElement.setAttribute("class", "text-secondary-dark");
+    abbrElement.appendChild(requiredText);
```

Comment:
IMO I think it's easier to keep track of changes across JS and ERB if we have everything in the DOM already but toggle attributes on them, like required, hidden, etc etc\

Is it possible to add these attributes in the ERB and hide them?